extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,164 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {addParamsToUrl, getSourceUrl} from '../../../src/url';
+import {elementByTag} from '../../../src/dom';
+import {getSocialConfig} from './amp-social-share-config';
+import {isLayoutSizeDefined, getLengthNumeral,
+  Layout} from '../../../src/layout';
+
+/** @const {number} */
+const DEFAULT_WIDTH = 60;
+
+/** @const {number} */
+const DEFAULT_HEIGHT = 44;
+
+/** @const {string} */
+const CONTAINER_TAG = 'span';
+
+/** @const {string} */
+const LINK_TAG = 'a';
+
+class AmpSocialShare extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FIXED;
+  }
+
+  /** @override */
+  buildCallback() {
+    /** @private @const {!Element} */
+    this.type_ = AMP.assert(this.element.getAttribute('type'),
+        'The type attribute is required. %s',
+        this.element);
+
+    /** @private @const {!Object} */
+    this.typeConfig_ = getSocialConfig(this.type_);
+
+    /** @private @const {!Object} */
+    this.config_ = this.getElementConfig_();
+
+    /** @private @const {number} */
+    this.width = getLengthNumeral(this.element.getAttribute('width'))
+      || DEFAULT_WIDTH;
+    /** @private @const {number} */
+    this.height = getLengthNumeral(this.element.getAttribute('height'))
+      || DEFAULT_HEIGHT;
+
+    this.renderShare_();
+  }
+
+  /**
+   * Renders the share based on the element config.
+   * @return {!Element}
+   */
+  renderShare_() {
+    const urlParams = {};
+
+    for (const param in this.typeConfig_['params']) {
+      const paramConf = this.typeConfig_['params'][param];
+      let paramValue = this.config_[param] || this.getDefaultValue_(
+          param, this.config_);
+      AMP.assert(!paramConf['required'] || paramValue !== null,
+        param + ' is a required attribute for ' + this.type_ + '. %s',
+        this.element);
+      if (paramValue == null && paramConf['type'] == 'fixed') {
+        paramValue = paramConf['value'];
+      }
+      if ('maxlength' in paramConf) {
+        const maxlength = paramConf['maxlength'];
+        AMP.assert(!paramValue || paramValue.length < maxlength,
+          param + ' cannot exceed ' + maxlength + '. %s', this.element);
+      }
+      if (paramValue != null) {
+        urlParams[paramConf['param']] = paramValue;
+      }
+    }
+
+    // Get the anchor or create one.
+    let link = elementByTag(this.element, LINK_TAG);
+    if (link === null) {
+      link = this.getWin().document.createElement(LINK_TAG);
+      link./*OK*/innerHTML = this.typeConfig_['text'];
+      link.setAttribute('target', '_blank');
```

Comment:
For `mailto:` you may need to do more in the case of iOS WebView. @mkhatib knows more.